### ISSUE TITLE
`/nexus-data/admin.password` gets re-created on Nexus restart.

### ISSUE THREAD

**Issue by AllRWeak** [sonatype/nexus-public#458]
When using Nexus Docker image `sonatype/nexus3:3.71.0` on initial install `/nexus-data/admin.password` file gets created. After logging in from the web console (or API) and changing the initial admin password the file gets deleted. When restarting the container the file gets re-created again although the admin password has been changed. This did not happen in previous versions is it a bug or an intended feature?

**Comment by PiR-B** [comment]
Hi,

Could you please let us know if this is intended behavior or not ?

Regards

**Comment by mrprescott** [comment]
Can you confirm if you're required to use the password in the re-created file to log in, or is your desired admin password still usable? (I realize this isn't your exact question.)

Can you also confirm if this is every time you restart, or only after restarting having done a database migration to H2?

**Comment by PiR-B** [comment]
Hi @mrprescott,

From my side I can confirm that the "re-generated" admin password is not usable.
From another side I can also confirm that every time I restart the service the `/opt/sonatype/sonatype-work/nexus3/admin.password` is regenerated.

I tried on a fresh install with nexus 3.71.0-06

Regards

**Comment by AllRWeak** [comment]
> Hi @mrprescott,
> 
> From my side I can confirm that the "re-generated" admin password is not usable. From another side I can also confirm that every time I restart the service the `/opt/sonatype/sonatype-work/nexus3/admin.password` is regenerated.
> 
> I tried on a fresh install with nexus 3.71.0-06
> 
> Regards

Yes exactly. I think this should not be the intended behavior that the `admin.password` gets regenerated even if it has been changed and the new one works.

**Comment by AllRWeak** [comment]
The same, I would call it, issue is also present in the latest `3.72.0` release.

**Comment by dhs-rec** [comment]
This is not restricted to the Docker images. I see re-appearing `admin.password` files after every update since 3.70.x.

**Comment by Joris29** [comment]
Can anyone confirm this issue is gone for 3.73.0 release?

**Comment by AllRWeak** [comment]
No, I confirm it's still present in `3.73.0`

**Comment by ChriFo** [comment]
The problem still existed in `3.75.1`.

**Comment by Joris29** [comment]
@mrprescott Is this something that will be looked into?

**Comment by ChriFo** [comment]
You can avoid the creation of `/nexus-data/admin.password` by setting `nexus.security.randompassword=false` in the file `nexus.properties`. [[StaticSecurityConfigurationSource.groovy#L143](https://github.com/sonatype/nexus-public/blob/main/components/nexus-security/src/main/java/org/sonatype/nexus/security/config/StaticSecurityConfigurationSource.groovy#L143)]

To avoid the then _simple_ default password in case of first run you can set your desired password with the env variable `NEXUS_SECURITY_INITIAL_PASSWORD`. [[StaticSecurityConfigurationSource.groovy#L60](https://github.com/sonatype/nexus-public/blob/main/components/nexus-security/src/main/java/org/sonatype/nexus/security/config/StaticSecurityConfigurationSource.groovy#L60)]

**Comment by dhs-rec** [comment]
Other than removing the file after every Nexus restart? No, I'm afraid there's none.

**Comment by ChriFo** [comment]
@Sammmmm @dhs-rec my suggestions are not working for you?

**Comment by dhs-rec** [comment]
Ah, sorry, didn't try.

**Comment by mbxsuite** [comment]
@ChriFo your hint to the source code was a good hint for using `NEXUS_SECURITY_RANDOMPASSWORD: false` in the docker compose environment of the nexus container. That alone is enough.

**Comment by monolithed** [comment]
I’m using version 3.85.0-03. After a week, all user passwords were reset. The container was not restarted. Does anyone know how to restore the admin password? I couldn’t find any up-to-date official instructions."

**Comment by ocontant** [comment]
---- EDITED ---- 
if you have an existing nexus-data volume with the admin.password file, and you rebuild Nexus, your environment variables will be ignored and Nexus will set the password to the admin.password value. 

Make sure you have no existing nexus-data volume when you set the environment variables:
    environment:
      - NEXUS_SECURITY_RANDOMPASSWORD=false
      - NEXUS_SECURITY_INITIAL_PASSWORD=admin

----------------


My pipelines are failing because Nexus set a random password and refuse the environment variables passed in Docker-Compose. 

nexus:
    image: sonatype/nexus3
    container_name: nexus
    ports:
      - "8081:8081"
    volumes:
      - nexus-data:/nexus-data
    environment:
      - NEXUS_SECURITY_RANDOMPASSWORD=false
      - NEXUS_SECURITY_INITIAL_PASSWORD=admin

    depends_on:
      nexus-init:
        condition: service_completed_successfully
